### PR TITLE
[java] Fixed inefficient use of keySet iterator instead of entrySet iterator…

### PR DESCRIPTION
**Pull request by epidemia**

### CODE REVIEW

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java [hunk 1]
```diff
@@ -49,9 +49,9 @@ public Object visit(ASTCompilationUnit node, Object data) {
                 bumpCounterFor(clazz);
             }
         }
-        for (String k : stats.keySet()) {
-            int val = stats.get(k);
-            Node n = nodes.get(k);
+        for (Map.Entry<String, Integer> entry : stats.entrySet()) {
+            int val = stats.get(entry.getKey());
```
Comment (oowekyala) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java]: ditto

File: pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symboltable/AcceptanceTest.java [hunk 2]
```diff
@@ -100,11 +100,10 @@ public void testDemo() {
...
         Map<NameDeclaration, List<NameOccurrence>> m = s.getDeclarations();
-        for (Iterator<NameDeclaration> i = m.keySet().iterator(); i.hasNext();) {
-            NameDeclaration d = i.next();
-            assertEquals("buz", d.getImage());
-            assertEquals("ArrayList", ((TypedNameDeclaration) d).getTypeImage());
-            List<NameOccurrence> u = m.get(d);
+        for (Map.Entry<NameDeclaration, List<NameOccurrence>> entry : m.entrySet()) {
+            assertEquals("buz", entry.getKey().getImage());
+            assertEquals("ArrayList", ((TypedNameDeclaration) entry.getKey()).getTypeImage());
+            List<NameOccurrence> u = m.get(entry.getKey());
```
Comment (oowekyala) [inline pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symboltable/AcceptanceTest.java]: This is wrong, should be `entry.getValue()`

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java [hunk 3]
```diff
@@ -49,9 +49,9 @@ public Object visit(ASTCompilationUnit node, Object data) {
...
+            Node n = nodes.get(entry.getKey());
```
Comment (oowekyala) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java]: ditto

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java [hunk 4]
```diff
@@ -98,9 +98,9 @@ void addField(String name, JavaFieldSignature sig) {
     @Override
     public boolean hasMatchingOpSig(String name, JavaOperationSigMask mask) {
         // Indexing on signatures optimises this type of request
-        for (JavaOperationSignature sig : operations.keySet()) {
-            if (mask.covers(sig)) {
-                if (operations.get(sig).contains(name)) {
+        for (Entry<JavaOperationSignature, Set<String>> entry : operations.entrySet()) {
+            if (mask.covers(entry.getKey())) {
+                if (operations.get(entry.getKey()).contains(name)) {
```
Comment (oowekyala) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java]: This is wrong, should be `entry.getValue()`

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java [hunk 5]
```diff
@@ -111,9 +111,9 @@ public boolean hasMatchingOpSig(String name, JavaOperationSigMask mask) {
 
     @Override
     public boolean hasMatchingFieldSig(String name, JavaFieldSigMask mask) {
-        for (JavaFieldSignature sig : fields.keySet()) {
-            if (mask.covers(sig)) {
-                if (fields.get(sig).contains(name)) {
+        for (Entry<JavaFieldSignature, Set<String>> entry : fields.entrySet()) {
+            if (mask.covers(entry.getKey())) {
+                if (fields.get(entry.getKey()).contains(name)) {
```
Comment (oowekyala) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java]: ditto

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java [hunk 6]
```diff
@@ -74,20 +74,20 @@ public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
 
         Map<VariableNameDeclaration, List<NameOccurrence>> vars = node.getScope()
                 .getDeclarations(VariableNameDeclaration.class);
-        for (VariableNameDeclaration decl : vars.keySet()) {
-            AccessNode accessNodeParent = decl.getAccessNodeParent();
-            if (vars.get(decl).isEmpty() || accessNodeParent.isTransient() || accessNodeParent.isStatic()) {
+        for (Map.Entry<VariableNameDeclaration, List<NameOccurrence>> entry : vars.entrySet()) {
```
Comment (oowekyala) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java]: I'd add a `VariableNameDeclaration decl = entry.getKey();` here for readability

**Comment by oowekyala** [comment]
FYI there's one checkstyle error too:
> [ERROR] src/test/java/net/sourceforge/pmd/lang/java/symboltable/AcceptanceTest.java:[12,8] (imports) UnusedImports: Unused import - java.util.Iterator.

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java (hunk 1 above)
Comment (epidemia) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java]: Fixed

File: pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symboltable/AcceptanceTest.java (hunk 2 above)
Comment (epidemia) [reply pmd-java/src/test/java/net/sourceforge/pmd/lang/java/symboltable/AcceptanceTest.java]: Fixed

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java (hunk 3 above)
Comment (epidemia) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/TooManyFieldsRule.java]: Fixed

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java (hunk 4 above)
Comment (epidemia) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/multifile/ClassStats.java]: Fixed

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java (hunk 6 above)
Comment (epidemia) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java]: Fixed

**Pull request merged**